### ISSUE TITLE
Error in opening block

### ISSUE THREAD

**Issue by liunx7594** [mozman/ezdxf#424]
I use the following code
```
import ezdxf

cad = ezdxf.readfile(r"test line.dxf")
msp = cad.modelspace()
obj = msp[0].virtual_entities()
```

![image](https://user-images.githubusercontent.com/55911351/116955756-07d78900-acc6-11eb-949d-160c62f3c5c0.png)

Get the error:
AttributeError: 'Text' object has no attribute 'virtual_entities'

**But strangely enough, I didn't use text.**

This block is made up of two half ellipses

I uploaded the DXF file.

[test insert.zip](https://github.com/mozman/ezdxf/files/6418816/test.insert.zip)

**Comment by mozman** [comment]
Can't reproduce your error with the provided test file. In your example you load "test line.dxf" instead of "test insert.dxf".

**Comment by liunx7594** [comment]
I'm very sorry, this situation is too strange, so I didn't make a correct judgment at the beginning.
I was testing the geographic information interface, and it turned out to be a problem.


Now, there is only one "insert" in my DXF file,
I use the "explode" command of AutoCAD2007, and the result is two half ellipses.


But when I use the following code.
```
import ezdxf
import ezdxf.addons.geo

cad = ezdxf.readfile(r'test insert.dxf')
msp = cad.modelspace()

all_from_insert = msp[0].explode()
for e in all_from_insert:
    print(e)
    print(e.get_points())
    ezdxf.addons.geo.proxy(e)
```

**There are two problems:** 
**1. The internal element of the block is actually "lwpolyline".**
**2. ezdxf.addons.geo.proxy reported error.**


The execution result of the original code is as follows
LWPOLYLINE(#152)
[(39482129.9462793, 3554328.753243976, 0.0, 0.0, 1.0), (39482129.95781776, 3554328.753243976, 0.0, 0.0, 1.0)]
Traceback (most recent call last):
  File "C:/Users/L/PycharmProjects/pythonProject/test_ezdxf_shapely_api.py", line 11, in <module>
    ezdxf.addons.geo.proxy(e)
  File "C:\Users\L\AppData\Local\Continuum\anaconda3\envs\py3.7\lib\site-packages\ezdxf-0.16.3b0-py3.7.egg\ezdxf\addons\geo.py", line 64, in proxy
    return GeoProxy.from_dxf_entities(entity, distance, force_line_string)
  File "C:\Users\L\AppData\Local\Continuum\anaconda3\envs\py3.7\lib\site-packages\ezdxf-0.16.3b0-py3.7.egg\ezdxf\addons\geo.py", line 354, in from_dxf_entities
    m = mapping(entity, distance, force_line_string)
  File "C:\Users\L\AppData\Local\Continuum\anaconda3\envs\py3.7\lib\site-packages\ezdxf-0.16.3b0-py3.7.egg\ezdxf\addons\geo.py", line 599, in mapping
    return _line_string_or_polygon_mapping(points, force_line_string)
  File "C:\Users\L\AppData\Local\Continuum\anaconda3\envs\py3.7\lib\site-packages\ezdxf-0.16.3b0-py3.7.egg\ezdxf\addons\geo.py", line 616, in _line_string_or_polygon_mapping
    raise ValueError('Invalid vertex count.')
ValueError: Invalid vertex count.


The document is as follows
[test insert.zip](https://github.com/mozman/ezdxf/files/6419533/test.insert.zip)

**Comment by mozman** [comment]
The `__geo_interface__` does not support circles, arc or bulges (which are arcs), and the `geo` add-on assumes a closed LWPOLYLINE should represent a 2D shape with more than 2 vertices (POLYGON). This is what the the error messages says. You have to preprocess the data or catch the `ValueError` and ignore such entities. The LWPOLYLINE in your block is "clever" way to create a full circle  by a closed LWPOLYLINE with 2 bulges, too clever for the `geo` add-on which always try to create a polygon from closed LWPOLYLINE entities.

**Comment by mozman** [comment]
The internal used `make_path()` function should flatten bulges, but fails on this special full circle LWPOLYLINE.

**Comment by mozman** [comment]
This is an floating point imprecision issue, the block defines a circle as a LWPOLYLINE of diameter 0.2, and insert this point at (39482129.952, 3554328.7532), the path tools considers the start and end point as close enough to be the same point and ignores the second point.

I will not fix this, because adding more tolerance makes it harder to to detect coincident points in normal cases. This is an unfortunate mix of big and small coordinates.

**Comment by liunx7594** [comment]
OK, thank you. I'll control it with exception

**Comment by mozman** [comment]
Added extended relative tolerance support to Vec2/3 to fix #431, which added the ability to increase the relative tolerance for polyline to path conversion to fix this issue.
